javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -14,24 +14,51 @@ package java.util
 
 import java.lang.Cloneable
 import java.lang.Utils._
+import java.util.ScalaOps._
 
 import scala.scalajs._
+import scala.scalajs.LinkingInfo.targetPureWasm
 
-class ArrayList[E] private (private[ArrayList] val inner: js.Array[E])
+// _size keeps the track of the effective size of the underlying scala.Array for Wasm
+// in JS, we can use innerJS.length for it's size.
+class ArrayList[E] private (private[ArrayList] var inner: AnyRef, private var _size: Int)
     extends AbstractList[E] with RandomAccess with Cloneable with Serializable {
   self =>
 
+  private val innerJS =
```

Comment:
We should add comments about these two fields similar to the ones we added in `ju.Random`:
https://github.com/scala-wasm/scala-wasm/blob/90e67d528f5facb5fdedf0839219c4434c8dd348/javalib/src/main/scala/java/util/Random.scala#L25-L34

Reply:
Also these two fields deserve explicit types.